src/gui/sdl_mapper.cpp

```diff
@@ -1200,6 +1202,94 @@ class CCHBindGroup : public CStickBindGroup {
 	uint16_t button_state = 0;
 };
 
+
+void MAPPER_TriggerEvent(const CEvent *event, const bool deactivation_state) {
+	assert(event);
+	for (auto &bind : event->bindlist) {
+		bind->ActivateBind(32767, true, false);
+		bind->DeActivateBind(deactivation_state);
+	}
+}
+
+class Typer {
+	public:
+		Typer() = default;
+		Typer(const Typer&) = delete; // prevent copy
+		Typer& operator=(const Typer&) = delete; // prevent assignment
+		~Typer() {
+			Stop();
+		}
+		void Start(std::vector<CEvent*>     *ext_events,
+		           std::vector<std::string> &ext_sequence,
+                   const uint32_t           wait_ms,
+                   const uint32_t           pace_ms) {
+			// Guard against empty inputs
+			if (!ext_events || ext_sequence.empty())
+				return;
+			Wait();
+			m_events = ext_events;
+			m_sequence = std::move(ext_sequence);
+			m_wait_ms = wait_ms;
+			m_pace_ms = pace_ms;
+			m_stop_requested = false;
+			m_instance = std::thread(&Typer::Callback, this);
```

Comment:
Back when Vogons mods were criticising us for suggesting C++11 they raised lots of stupid reasons (and they did not acknowledge our answers) and a single good reason. That good reason was a problem with `std::thread` when using MinGW - some time ago the implementation was not finished and had some issues with wrapping underlying OS threading API (It won't be a problem for our Visual Studio builds or any other platform except MinGW). I don't remember if that problem was finally resolved in MinGW or not. If it'll turn out it's not solved still - we'll need to go back to using `SDL_Thread` :(

If that'll be the case, then perhaps the good solution would be to create a wrapper implementing the same interface as `std::thread`, but using SDL_Thread abstraction underneath (don't jump into implementing it! I do have similar wrapper lying around somewhere (not targetting SDL_Thread, but a different API, but adaptation might be somewhat easy).

*another place where Rust would be better than C++ :(*

Reply:
Argh.. thanks for the explanation and knowing you've got a wrapper in your pocket :-)
Will do some testing on the VMs and physical machines I have!

Reply:
Builds and runs without issue using today's mingw packages for gcc 9.3 and clang 9.x (the default installations for both compilers).  Setting to resolved for now, can reopen if there are other issues.

Reply:
Soo, I am not going to pretend like I understand all the nuances here; some relevant StackOverflow discussions:

- https://stackoverflow.com/questions/17242516/mingw-w64-threads-posix-vs-win32/30390278#30390278
- https://stackoverflow.com/questions/37358856/does-mingw-w64-support-stdthread-out-of-the-box-when-using-the-win32-threading

I am very comfortable with using `std::thread` on Linux, macOS, and on Windows with MSVC. I am still not comfortable with using it on MinGW. I don't even fully understand all the versions of MinGW that user can install nor how user can select or switch threading model in there…

If we're switching to `std::thread`, then what do we do with SDL threading API we already use in code?

Reply:
It seems like there were nuances back in those 2016 and 2015 questions; but we're 4+ year past that now, and the currently-supported mingw compilers (gcc, clang) build `sdl_mapper.cpp` without warnings, and the binaries run without threading issues (no addon threading packages or tweaks needed).

If someone hits an issue building `std::thread`, then I am happy to step down to SDL's threading; but my guess is these cases will be rare or non-existent (given we already require `-std=c++11`. 

> If we're switching to std::thread, then what do we do with SDL threading API we already use in code?

This PR adds a feature using C++11 best practices - and to me, that means using the languages in-built threading, and not using OS-library calls.  So I'm not sure we need to answer this question (just in the same way we don't ask that question about other C++11 improvements we've made to the code; there are still lots of code using non-C++11 constructs).  

I think the only other area using threading is the CD-DA decode callback to use SDL's mixer. We can migrate that in another PR (I hadn't thought of it; thanks!).  That will let us eliminate the use of `SDL_thread.h` and lighten the code a bit more.

Reply:
OK, let's not worry about this prematurely then :) We just need to make sure not to let various threading APIs uncontrollably propagate through the codebase.

Reply:
> We just need to make sure not to let various threading APIs uncontrollably propagate through the codebase.

:+1: , fully agree. Next change to `cdrom_image.cpp` will be to move it to c++11 threading, as well, and we'll be done with threading APIs entirely :-)